src/sentry/rules/actions/base.py

```diff
@@ -55,29 +63,45 @@ class IntegrationEventAction(EventAction):
     Intermediate abstract class to help DRY some event actions code.
     """
 
-    def is_enabled(self):
-        return self.get_integrations().exists()
+    @property
+    def provider(self) -> str:
+        raise NotImplementedError
 
-    def get_integration_name(self):
-        """
-        Get the integration's name for the label.
+    @property
```

Comment:
Are these functional changes? Not sure what's going on here.

Reply:
`IntegrationEventAction` is an abstract class and it's trying to reference `self.provider`, `self.integration_key`, and `self.ticket_type` which are supposed to be implemented by the child classes.
I thought mypy wouldn't be able to infer the types without defining them here but I just checked and it's not complaining. I'll move this change to another PR.

Reply:
Oops nvm, mypy _is_ actually complaining. It's fixed in https://github.com/getsentry/sentry/pull/32500 so I'll rebase onto that branch.

Reply:
@evanh I had to leave those empty abstract properties in order to fix mypy. Is it OK like this or would you prefer I wait until after https://github.com/getsentry/sentry/pull/32500 lands?